.github/workflows/linux.yml

```diff
@@ -173,6 +177,11 @@ jobs:
           path: target/cargo-timings-*
       - name: Build mach package
         run: ./mach package --${{ inputs.profile }}
+      - name: Generate mach package artifact attestation
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-name: 'linux'
+          subject-path: 'target/${{ inputs.profile }}/servo-tech-demo.tar.gz'
```

Comment:
Does this actually works on PRs? In mozjs we have separated "elevated" workflows for pushes and unelevated for rest: https://github.com/servo/mozjs/pull/501

Also per https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds#what-to-attest we should not attestate "frequent builds that are just for automated testing", although I personal see no problem in signing all builds (this simplifies CI), but maybe GitHub doesn't want us to spam with attestation everywhere. Maybe we should use `if: ${{ inputs.upload }}` for such steps.

cc @jschwe, who pushed for attestation before.

Reply:
`if: ${{ inputs.upload }}` seems reasonable